codec_default.go

```diff
@@ -0,0 +1,565 @@
+package avro
+
+import (
+	"encoding"
+	"encoding/binary"
+	"fmt"
+	"math/big"
+	"reflect"
+	"unsafe"
+
+	"github.com/modern-go/reflect2"
+)
+
+func createDefaultDecoder(cfg *frozenConfig, schema Schema, def any, typ reflect2.Type) ValDecoder {
```

Comment:
This is not really needed. An example of this in the encoder can be found here and IMO is much simpler: https://github.com/hamba/avro/blob/main/codec_record.go#L174

Reply:
It’s unclear to me how this example can help in this case.